bin/runtest.ml

```diff
@@ -27,39 +28,128 @@ let runtest_info =
         ; ( "Run tests sequentially without output buffering"
           , "dune runtest --no-buffer -j 1" )
         ; "Run tests in a specific build context", "dune runtest _build/my_context/"
+        ; "Run inline tests of a library", "dune runtest src/my_lib_with_tests"
         ]
     ]
   in
   Cmd.info "runtest" ~doc ~man ~envs:Common.envs
 ;;
 
-let find_cram_test path ~parent_dir =
+let find_cram_tests_in_dir (dir : Path.Source.t) =
   let open Memo.O in
-  Source_tree.nearest_dir parent_dir
+  Source_tree.nearest_dir dir
   >>= Dune_rules.Cram_rules.cram_tests
   (* We ignore the errors we get when searching for cram tests as they will
-     be reported during building anyway. We are only interested in the
-     presence of cram tests. *)
+     be reported during building anyway. We are only interested in the presence
+     of cram tests. *)
   >>| List.filter_map ~f:Result.to_option
-  (* We search our list of known cram tests for the test we are looking
-     for. *)
-  >>| List.find ~f:(fun (test : Source.Cram_test.t) ->
+;;
+
+let find_cram_test (path : Path.Build.t) =
+  let open Memo.O in
+  let path = Path.Build.drop_build_context_exn path in
+  let parent_dir = Path.Source.parent_exn path in
+  find_cram_tests_in_dir parent_dir
+  >>| List.find_map ~f:(fun (test : Source.Cram_test.t) ->
     let src =
       match test with
       | File src -> src
       | Dir { dir = src; _ } -> src
     in
-    Path.Source.equal path src)
+    Option.some_if
+      (Path.Source.equal path src)
+      (`Test (Path.source parent_dir, Source.Cram_test.name test)))
+;;
+
+let find_inline_tests_in_dir dir =
+  let open Memo.O in
+  Dune_rules.Dune_load.stanzas_in_dir dir
+  >>= Memo.Option.map ~f:(fun dune_file ->
+    Dune_file.find_stanzas dune_file Dune_rules.Library.key
+    >>| List.filter_map ~f:(fun { Library.sub_systems; name = _loc, name; _ } ->
+      Option.some_if
+        (Dune_rules.Sub_system_name.Map.mem
```

Comment:
Using the sub system is a bit too level for functionality like this. I think you should expose what you need from the `Dune_rules.Inline_tests` module.